app/models/project.rb

```diff
@@ -664,27 +666,25 @@ def check_status
     StructuredLog.capture("CHECK_STATUS_CHANGE", { platform: platform, name: name, status_code: response.response_code }) if platform.downcase == "npm"
 
     if platform.downcase == "packagist" && [302, 404].include?(response.response_code)
-      update_attribute(:status, "Removed")
+      update(status: "Removed", status_reason: "Response #{response.response_code}")
```

Comment:
note that `update_attribute` wasn't running validations, but I don't see a big reason to not run them.